policy/resource.go

```diff
@@ -136,23 +195,38 @@ func (r Resource) ValidateBucket(bucketName string) error {
 
 // parseResource - parses string to Resource.
 func parseResource(s string) (Resource, error) {
-	if !strings.HasPrefix(s, ResourceARNPrefix) {
-		return Resource{}, Errorf("invalid resource '%v'", s)
+	r := Resource{Type: unknownARN}
+	switch {
+	case strings.HasPrefix(s, ResourceARNPrefix):
+		r.Pattern = strings.TrimPrefix(s, ResourceARNPrefix)
+		r.Type = ResourceARNS3
+
+	case strings.HasPrefix(s, ResourceARNKMSPrefix):
+		r.Pattern = strings.TrimPrefix(s, ResourceARNKMSPrefix)
+		r.Type = ResourceARNKMS
+	default:
+		return r, Errorf("invalid resource '%v'", s)
```

Comment:
Simpler and not something we need to maintain:

```suggestion
	for k, v := range ARNPrefixToType {
		if rem, ok := strings.CutPrefix(s, k); ok {
			r.Type = v
			r.Pattern = rem
			break
		}
	}
	if r.Type == unknownARN {
		return r, Errorf("invalid resource '%v'", s)
```